read-fonts/src/tables/postscript/string.rs

```diff
@@ -0,0 +1,504 @@
+//! PostScript string identifiers.
+
+/// PostScript string identifier (SID).
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+pub struct StringId(u16);
+
+impl StringId {
+    pub fn new(raw: u16) -> Self {
+        Self(raw)
+    }
+
+    /// Resolves the identifier as a standard string.
+    ///
+    /// If the identifer represents a standard string, returns `Ok(string)`,
+    /// otherwise returns `Err(index)` with the index that should be used to
+    /// retrieve the string from the CFF string INDEX.
+    ///
+    /// The standard string set is available in the section
+    /// "Appendix A - Standard Strings" at <https://adobe-type-tools.github.io/font-tech-notes/pdfs/5176.CFF.pdf>.
+    pub fn standard_string(self) -> Result<Latin1String<'static>, usize> {
```

Comment:
Curious about how this API ends up being used, it feels like it's mixing concerns a little bit? I would be tempted to split this up into separate methods, like

```rust
fn is_standard_string(&self) -> bool;
fn standard_string(&self) -> Option<Latin1String<'static>>;
fn index(&self) -> Option<usize>;
```

or something along these lines? And depending on what the types look like you could maybe move the whole "standard or index" functionality into a method on whatever type encapsulates the string storage?

Reply:
This was just for economy of expression. If you have one of these things, you either want the standard string or the index to look up the custom one.  In this case, `Result` seemed like a nice option to represent “either” similar in style to the `binary_search` methods 🤷🏻‍♂️

In most cases, the user will just pass this id to some `string()` method on the CFF table which will do all this internally.

Reply:
Going to go ahead and merge this as-is because it is a good match for the current use cases. We can revisit if the API proves cumbersome for future uses.